src/packages/solid/components/Button/Button.tsx

```diff
@@ -40,6 +41,10 @@ export interface ButtonStyleProps {
 }
 
 const Button: Component<ButtonProps> = props => {
+  
+  const [prefixCheckbox, prefixIcon] = splitProps(props.prefix, ['icon', 'checked']);
+  const [suffixCheckbox, suffixIcon] = splitProps(props.suffix, ['icon', 'checked']);
```

Comment:
We're trying to avoid using splitProps as that uses JS proxies which won't work on older browsers. Can we do this another way?
props.prefix.icon and props.prefix.checkbox? Then you just spread them.

Reply:
im really just curious - there's no mention of proxies here so i thought we were potentially safe 😬 https://www.solidjs.com/docs/latest/api#splitprops but it sounds like probably not?